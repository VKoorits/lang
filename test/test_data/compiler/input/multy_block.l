var a = 7
while a < 10:
	if a % 2:
		a = a + 1
	else:
		a = a - 3
