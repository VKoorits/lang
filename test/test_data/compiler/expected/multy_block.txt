CONSTANTS:
	1: 7
	2: 10
	3: 2
	4: 1
	5: 3
CODE:
1:	NEW_VAR 1
2:	PUSH 1
3:	PUSH_CONST 1
4:	STORE
5:	PUSH 1
6:	PUSH_CONST 2
7:	LESS
8:	JUMP_IF_NOT 0x126
9:	NEW_VAR 0
10:	PUSH 1
11:	PUSH_CONST 3
12:	MOD
13:	JUMP_IF_NOT 0x89
14:	NEW_VAR 0
15:	PUSH 1
16:	PUSH 1
17:	PUSH_CONST 4
18:	BINARY_ADD
19:	STORE
20:	DELETE_VAR 0
21:	GOTO 0x116
22:	NEW_VAR 0
23:	PUSH 1
24:	PUSH 1
25:	PUSH_CONST 5
26:	BINARY_SUB
27:	STORE
28:	DELETE_VAR 0
29:	DELETE_VAR 0
30:	GOTO 0x20
31:	DELETE_VAR 1
32:	EXIT
